test/integration/nse_dataplane_heal_test.go

```diff
@@ -0,0 +1,94 @@
+package nsmd_integration_tests
+
+import (
+	"fmt"
+	"github.com/networkservicemesh/networkservicemesh/test/integration/nsmd_test_utils"
+	"github.com/networkservicemesh/networkservicemesh/test/kube_testing/pods"
+	"testing"
+	"time"
+
+	"github.com/networkservicemesh/networkservicemesh/test/kube_testing"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+func TestDataplaneHealLocal(t *testing.T) {
+	RegisterTestingT(t)
+
+	if testing.Short() {
+		t.Skip("Skip, please run without -short")
+		return
+	}
+
+	testDataplaneHeal(t, 1)
+}
+
+func TestDataplaneHealRemote(t *testing.T) {
+	RegisterTestingT(t)
+
+	if testing.Short() {
+		t.Skip("Skip, please run without -short")
+		return
+	}
+
+	testDataplaneHeal(t, 2)
+}
+
+/**
+If passed 1 both will be on same node, if not on different.
+*/
+func testDataplaneHeal(t *testing.T, nodesCount int) {
+	k8s, err := kube_testing.NewK8s()
+	defer k8s.Cleanup()
+
+	Expect(err).To(BeNil())
+
+	s1 := time.Now()
+	k8s.Prepare("nsmd", "nsc", "nsmd-dataplane", "icmp-responder-nse", "jaeger")
+	logrus.Printf("Cleanup done: %v", time.Since(s1))
+
+	// Deploy open tracing to see what happening.
+	jaeger := k8s.CreatePod(pods.Jaeger())
```

Comment:
We do not want to test `Jaeger` deployments in our CI. If you need these during the development time, please use env variables toe enable them.

Reply:
Since we have this inside main NSMD codebase, for using Jarger, I think we need to be sure it all working with/without it. So we should have at least some small test with it. I've planed to access API to it, to retrieve a full list of operations performed during Connection establish/heal etc, but not yet able to do so. So lets pod's definition in tests to be here, and use it in some other PR.